shared/base/src/commonMain/kotlin/kmp/shared/analytics/domain/model/AnalyticsEvent.kt

```diff
@@ -0,0 +1,25 @@
+package kmp.shared.analytics.domain.model
+
+/**
+ * Class defining a analytics event that can be logged.
+ *
+ * @param eventName The type (name) of this event
+ * @param parameters Parameters for this event
+ */
+class AnalyticsEvent(
+    val eventName: AnalyticName.Event,
+    val parameters: Map<AnalyticName.Param, String> = emptyMap(),
+)
+
+/**
+ * Names used to identify certain action reported to analytics
+ */
+object AnalyticName {
+    enum class Event(val eventName: String) {
+        ToastPresented("toast_presented"),
+    }
+
+    enum class Param(val parameterName: String) {
+        PresentedFrom("presented_from"),
+    }
+}
```

Comment:
This could potentially contain a lot of enum cases.. Maybe we should think of some way how to define analytic events in separate files (by features, for example)? What do you think?